server/src/main/java/com/defold/extender/ExtenderUtil.java

```diff
@@ -287,10 +280,108 @@ static String getAppManifestContextString(AppManifestConfiguration manifest, Str
+        List<String> items = new ArrayList<>();
+        if (l1 != null) {
+            items.addAll(l1);
+        }
+        if (l2 != null) {
+            items.addAll(l2);
+        }
+        return items;
+    }
+
+    private static final String MERGE_KEY_REPLACE = "_replace";
+    static private boolean isMergeOp(String name)
+    {
+        if (name.endsWith(MERGE_KEY_REPLACE))
+            return false;
+        return true;
+    }
+
+    static private String stripMergeKey(String name)
+    {
+        if (name.endsWith(MERGE_KEY_REPLACE))
+            return name.substring(0, name.length()-MERGE_KEY_REPLACE.length());
+        return name;
+    }
+
+    // Copies the original context, and appends the extra context's elements, if the keys and types are valid
+    static public Map<String, Object> mergeContexts(Map<String, Object> originalContext, Map<String, Object> extensionContext) throws ExtenderException {
+        Map<String, Object> context = new HashMap<>(originalContext);
+
+        // Clean the names of the previous context
+        Set<String> originalKeys = new HashSet<>(context.keySet());
+        for (String k : originalKeys) {
+            String key = stripMergeKey(k);
+            if (!key.equals(k))
+            {
+                Object v = context.get(k);
+                context.remove(k);
+                context.put(key, v);
+            }
+        }
+
+        Set<String> keys = extensionContext.keySet();
+        for (String k : keys) {
+
+            boolean isMergeOp = isMergeOp(k);
+            String key = stripMergeKey(k);
+
+            Object v1 = context.getOrDefault(key, context.getOrDefault(k, null));
+            Object v2 = extensionContext.get(k);
+
+            if (v1 == null && v2 == null) {
+                // Simply skip keys that hold no values at all
+                context.remove(key);
+                continue;
+            }
+
+            if (v1 != null && v2 != null && !v1.getClass().equals(v2.getClass())) {
+                throw new ExtenderException(String.format("Wrong manifest context variable type for %s: Expected %s, got %s: %s", k, v1.getClass().toString(), v2.getClass().toString(), v2.toString()));
+            }
+            if (v2 != null && v2 instanceof List && !ExtenderUtil.isListOfStrings((List<Object>) v2)) {
+                throw new ExtenderException(String.format("The context variables only support lists of strings. Got %s (type %s)", v2.toString(), v2.getClass().getCanonicalName()));
+            }
+
+            if (v1 != null && v2 != null && v1 instanceof List) {
+                //System.out.printf("merge op %s : %s  + %s  (merge: %s)\n", key, v1.toString(), v2.toString(), isMergeOp?"true":"false");
+
+                if (isMergeOp) {
+                    v1 = ExtenderUtil.mergeLists((List<String>) v1, (List<String>) v2);
+                } else {
+                    v1 = v2;
+                }
```

Comment:
If the property name ends with "_replace", the else-path here is taken, and we keep the v2 value as-is.

If not, we use the backwards compatible way of merging the two lists v1 and v2.